crates/sui-json-rpc/src/name_service.rs

```diff
@@ -128,14 +130,16 @@ pub enum DomainParseError {
     InvalidUnderscore,
     #[error("Domain must contain at least one label")]
     LabelsEmpty,
+    #[error("Domain must include only one separator")]
+    InvalidSeparator,
 }
 
 impl FromStr for Domain {
     type Err = DomainParseError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         /// The maximum length of a full domain
-        const MAX_DOMAIN_LENGTH: usize = 200;
+        const MAX_DOMAIN_LENGTH: usize = 253;
```

Comment:
Curious: how is the max domain length determined, is this a standard value?

Reply:
This is a good question and it's an open one on the SC side too.
@bmwill I was planning to ask you when you're back (since we're not going live with it until Jan), if there's a reason we chose 200 instead of 253.
https://en.wikipedia.org/wiki/Domain_Name_System

Reply:
Discussed with brandon about it. We needed some extra space to support the `sui.id`. Ended up going with 235 both here and on-chain.